tests/integration/utils/named-list.test.ts

```diff
@@ -0,0 +1,120 @@
+import { expect, test, describe } from "vitest";
+import { render, screen } from "@testing-library/svelte";
+import NamedList from "$lib/components/composites/utils/NamedList.svelte";
+// @ts-expect-error "Snippets have implicetly type any"
+import { listItemComponent, listItemSkeleton } from "./ExampleSnippets.svelte";
+
+describe("NamedListComponent", () => {
+    test("When all required props are provided, then the named list is completely shown.", async () => {
+        const componentData: Promise<string[]> = Promise.resolve(
+            Array.from({ length: 15 }, (_, i) => `Hello world ${i}`),
+        );
+
+        render(NamedList, {
+            props: {
+                listName: "Test List",
+                items: componentData,
+                listItemComponent: listItemComponent,
+                listItemSkeleton: listItemSkeleton,
+            },
+        });
+
+        // List title is shown
+        expect(screen.getByText("Test List")).toBeInTheDocument();
+
+        setTimeout(() => {
```

Comment:
I'm not sure if this is executed because the method doesn't wait for the timeout to be finished.

AFAIK, you can use `waitFor` for this.

The same applies for the other tests.